### PR TITLE
Mark question as spam

### PR DESCRIPTION
**Pull request by faityworld**

### CODE REVIEW

**Review by zsoltime — CHANGES_REQUESTED** [review]
There is no interaction with the database. You should add a new field, something like `spam` or `flagged_by` to the `Question` schema. It could be an array with a list of user IDs that marked the question as spam. The post request should contain a user ID in the request body. You can use `findOneAndUpdate`. If the array contains the user ID, you can pull it; if not, you can push it.

Or maybe you can use `$addToSet` to solve this.

**Comment by faityworld** [comment]
Instead of user's data was created a user's object inside the controller. It should be changed on a user's session data after we add authentication.

**Review by zsoltime — CHANGES_REQUESTED** [review]
Could you check out #71, please? I think something like that should work.

File: server/controller/index.js [hunk 1]
```diff
@@ -57,3 +57,32 @@ exports.getId = async (ctx) => {
...
+
+
+exports.markSpam = async (ctx) => {
+  ctx.user = { id: '5a2ff257fc13ae6d59000535' };
+  if (!ctx.user) {
+    ctx.body = 'Not authourized';
+    return ctx.body;
+  }
+  await Question.findOne({ _id: ctx.params.id})
+    .then(async (data) => {
```
Comment (Heyjp) [inline server/controller/index.js]: You don't need to include `async` here the function `markSpam` is already asynchronous.

File: server/controller/index.js [hunk 2]
```diff
@@ -57,3 +57,32 @@ exports.getId = async (ctx) => {
...
+        return ctx.body;
+      }
+      const userMarked = data.spam.includes(ctx.user.id);
+      const updates = userMarked
+        ? { $pull: { spam: ctx.user.id } }
+        : { $addToSet: { spam: ctx.user.id } };
+
+      await Question.update({ _id: ctx.params.id }, updates)
+        .then((res) => {
+          ctx.body = res;
```
Comment (Heyjp) [inline server/controller/index.js]: Can you return `ctx.body` here.

File: server/controller/index.js [hunk 3]
```diff
@@ -57,3 +57,32 @@ exports.getId = async (ctx) => {
...
+        }).catch((err) => {
+          ctx.body = err;
```
Comment (Heyjp) [inline server/controller/index.js]: Also return here.

File: server/controller/index.js [hunk 4]
```diff
@@ -57,3 +57,32 @@ exports.getId = async (ctx) => {
...
+        });
+    }).catch((err) => {
+      ctx.body = err;
```
Comment (Heyjp) [inline server/controller/index.js]: Last return for `ctx.body` here.

File: server/controller/index.js [hunk 5]
```diff
@@ -57,3 +57,32 @@ exports.getId = async (ctx) => {
       return ctx.body;
     });
 };
+
+
+exports.markSpam = async (ctx) => {
+  ctx.user = { id: '5a2ff257fc13ae6d59000535' };
```
Comment (Heyjp) [inline server/controller/index.js]: We will be using `koa-passport` package later. So I believe user information will be saved as `ctx.state.user` could you update the function to use a single const with information saved from `ctx.state.user.id`.

**Review by Heyjp — CHANGES_REQUESTED** [review]

**Comment by Heyjp** [comment]
Function looks good, just needs a few small updates. But good job @OStefani  👍

File: server/controller/index.js (hunk 1 above)
Comment (faityworld) [reply server/controller/index.js]: It should be here in order we could get a response from db.

File: server/controller/index.js [hunk 6]
```diff
@@ -58,31 +58,20 @@ exports.getId = async (ctx) => {
...
-        .then((res) => {
-          ctx.body = res;
-        }).catch((err) => {
-          ctx.body = err;
-        });
-    }).catch((err) => {
-      ctx.body = err;
-    });
+  const doc = await Question.findOne({ _id: ctx.params.id })
+    .then((data) => data )
```
Comment (Heyjp) [inline server/controller/index.js]: I made a mistake, you can remove the parentheses so `.then(data => data)` is fine

File: server/controller/index.js [hunk 7]
```diff
@@ -57,3 +57,21 @@ exports.getId = async (ctx) => {
...
 };
+
+exports.markSpam = async (ctx) => {
+  const user = ctx.state.user.id;
+  if (!user) {
+    ctx.body = 'Not authorized';
+    return ctx.body;
+  }
+  const doc = await Question.findOne({ _id: ctx.params.id })
+    .then((data) => data )
```
Comment (Heyjp) [inline server/controller/index.js]: You can remove the parentheses around data. ESLint allows `.then(data => data)` apologies for the mistake.

**Review by Heyjp — CHANGES_REQUESTED** [review]

**Comment by Heyjp** [comment]
We also have branch conflicts, would you be able to resolve them as well? Otherwise I can give it a go 😅

**Comment by Heyjp** [comment]
Can you revert back to before you made the merge. We to keep the existing code in development branch but just add your route + controller function. Your latest commit removed a lot of the previous code.

**Pull request closed without merging**